dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -979,10 +981,67 @@ public static int runCmd(String[] cmd, File workingDirectory, File outputFile) t
         pb.redirectErrorStream(true);
 
         Log.info(c, "runCmd", "Running command " + Arrays.asList(cmd));
-        Process p = pb.start();
-        int exitCode = p.waitFor();
-        return exitCode;
+        
+        int  hardTimeout = Integer.parseInt(System.getProperty("fat.timeout", "10800000"));
+        long softTimeout = -1;
+
+        // We need to ensure that the hard timeout is large enough to avoid future issues
+        if (hardTimeout >= 30000) {
+            softTimeout = hardTimeout - 15000; // Soft timeout is 15 seconds less than hard timeout
+        }
 
+        Process p = pb.start();
+        int returnCode = 1;
+        boolean returnStatus;
+        if (softTimeout > -1) {
+            returnStatus = p.waitFor(softTimeout, TimeUnit.MILLISECONDS);  // Requires Java 8+
+            if (returnStatus == false) {
+                // Parse through the mvn logs
+                if (outputFile.exists() && outputFile.canRead()) {
+                    try (Scanner s = new Scanner(outputFile)) {
+                        // Get the last few lines from the MVN log
+                        ArrayList<String> lastLines = new ArrayList<String>();
+                        int numOfLinesToInclude = 7;  // We will include the last 7 lines of the output file in the timeout message
+                        while (s.hasNextLine()) {
+                            if (lastLines.size() < numOfLinesToInclude) {
+                                lastLines.add(s.nextLine());
+                            } else {
+                                lastLines.remove(0);
+                                lastLines.add(s.nextLine());
+                            }
+                        }
+                        
+                        // Prepare the timeout message
+                        String timeoutMsg = "Timeout occurred. FAT timeout set to: " + hardTimeout + "ms (soft timeout set to " + softTimeout + "ms). The last " +
+                                            numOfLinesToInclude + " lines from the mvn logs are as follows:\n";
+                        for (String line : lastLines) {
+                            timeoutMsg += line + "\n";
```

Comment:
does each line of mvn output have a `\n` char at the end already?

Reply:
Checked locally and it appears there isn't \n